### ISSUE TITLE
Advanced learning rate settings in 'Using DIGITS ...' tutorial ?

### ISSUE THREAD

**Issue by JVR32** [NVIDIA/DIGITS#1022]
Hello,

I am following the tutorial on 'Using DIGITS to train Object Detection network'.
[https://github.com/NVIDIA/DIGITS/tree/master/examples/object-detection]

I downloaded and prepared the KITTI data as explained in the document. Then I built a New Object Detection Dataset. That succeeded. Only difference that I see, is that in the tutorial they speak about 5984 images in the training set and 1496 images in the validation set ; in my case, there are 6373 images in the training set and 1108 images in the validation set. But I suppose this is a minor detail.

I tried to create a model using the settings from the document, but the graph I get during training is not the same as in the example, so I stopped after a number of epochs.

![graph](https://cloud.githubusercontent.com/assets/19925967/18122122/30d8e22c-6f68-11e6-80ec-b03f9e37f7f2.jpg)

I used all the settings from the document, so I put 0.0001 for the base learning rate. But I didn't change the advanced settings : 
-> policy : step down
-> step size : 33%
-> gamma : 0.1

So I was wondering which (advanced) settings were used in the tutorial? I don't see them specified in the document, so I would like to know if the standard settings were used (like I did), or if another option (e.g. exponential decay with certain gamma, ...) was selected for getting the graph that we can see in the tutorial.

With kind regards.

**Comment by lukeyeager** [comment]
Actually, that graph doesn't look so bad. Sometimes the loss will jump up really high, but then go back down and keep behaving normally. It's hard to tell, but it looks like your mAP is actually pretty high (~50% is what you're shooting for). Click on the text that says "loss_bbox (train)" to hide it - that will make your accuracy graphs stand out more clearly.

**Comment by JVR32** [comment]
Yesterday, I restarted the same experiment. This is the graph after 15 epochs : 

![graph2](https://cloud.githubusercontent.com/assets/19925967/18158220/2a5e949a-7022-11e6-83a6-b6f5f713014b.jpg)

I asked 30 epochs (as in the tutorial), but it seems to me that the accuracy is quite stable (between 0.1 and 0.2) at this moment ; I don't expect it to improve significantly in the last 15 epochs.

Executing the experiment with the same settings as in the tutorial should produce -more or less- the same results I guess ...
That's why I'm wondering which 'advanced settings' were used in the tutorial for the learning rate. I have a feeling that it wasn't ...

> -> base learning rate : 0.0001
> -> policy : step down
> -> step size : 33%
> -> gamma : 0.1

... , the settings I'm using right now.

I'm looking for the differences between my experiment and the one in the tutorial, and according to me, the advanced setttings for the learning rate are most likely candidates ...

**Comment by lukeyeager** [comment]
Don't worry so much about the shape of the graph. Here are a few [relatively] successful training runs from my machine - the graph may jump around a bit before converging. What you should look for is an mAP value converging to ~50%.

<img src="https://cloud.githubusercontent.com/assets/687269/18175546/8debf8e6-7026-11e6-9281-10b698d5df1e.jpg" height=150>
<img src="https://cloud.githubusercontent.com/assets/687269/18175547/8dedb488-7026-11e6-9d36-c51d3d2a59fc.jpg" height=150>
<img src="https://cloud.githubusercontent.com/assets/687269/18175545/8deb8668-7026-11e6-81f9-02202b4dab96.jpg" height=150>

And yes, I'm using the same learning rate settings as you.

**Comment by JVR32** [comment]
I'm not worried about the shape of the graph...
If you take a look at your graphs, you can see that after 15 epochs the accuracy is >0.5 and still improves a bit after that. In the 2 experiments that I started, I reach an accuracy between 0.1 and 0.2 after 15 epochs, the level that is already reached after 6 epochs. So I will run the experiment from the tutorial once again, and check the accuracy after 30 epochs, but I'm afraid the accuracy will remain between 0.1 and 0.2. But you never know ... (I'll keep you posted)

I understand that the shapes of the curves can be different, but I would expect that the final accuracy should be -more or less- comparable, and that is the case in your graphs. I literally followed the tutorial but still I got a different result (reasonable difference), so I must be overlooking something ...

But at this moment, I have no idea what that can be.

**Comment by lukeyeager** [comment]
Hover over the points in your graph and check the values.

**Comment by JVR32** [comment]
Yes, I see what you mean now!

Yesterday I followed your advice : 
> Click on the text that says "loss_bbox (train)" to hide it - that will make your accuracy graphs stand out more clearly.

I didn't see a big difference then, but now -if I hide the boxes about 'loss_coverage'- , I see the graphs that I expected : 

![graph](https://cloud.githubusercontent.com/assets/19925967/18186175/a14a912e-70a1-11e6-9d77-7a51dfd4d34c.jpg)

I totally misinterpreted the scale at the right side of the graphs. I wrongly assumed the 0.2 was also pointing to the validation curves.

Thank you very much Luke for clearing this out!
